src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -1271,16 +1281,23 @@ private void Abort(Exception abortException)
 
         private void AbortStreams(int lastValidStream, Exception abortException)
         {
-            bool shouldInvalidate = false;
-            lock (SyncObject)
+            Debug.Assert(lastValidStream >= 0);
+            bool hasAnyActiveStream = false;
+
+            if (_lastStreamId == -1)
+            {
+                _lastStreamId = lastValidStream;
```

Comment:
This doesn't seem quite right. Per RFC:

```
   An endpoint MAY send multiple GOAWAY frames if circumstances change.
   For instance, an endpoint that sends GOAWAY with NO_ERROR during
   graceful shutdown could subsequently encounter a condition that
   requires immediate termination of the connection.  The last stream
   identifier from the last GOAWAY frame received indicates which
   streams could have been acted upon.  Endpoints MUST NOT increase the
   value they send in the last stream identifier, since the peers might
   already have retried unprocessed requests on another connection.
```

That said, we weren't handling this properly before, so if you want to file a separate issue, go ahead.

Reply:
(Nevermind, I missed the logic below. Seems fine to me.)

Reply:
@geoffkizer do you mean not checking the error flags?

Reply:
Current code does not increase last stream identifier (only decreasing is allowed due to Math.Min)

Reply:
Yeah, it's fine. Just ignore this comment.